python/Ganga/test/GPI/FileTests/TestMassStorageClient.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import absolute_import
+from Ganga.testlib.GangaUnitTest import GangaUnitTest
+
+import tempfile
+import datetime
+import time
+import random
+import os
+import shutil
+import copy
+import string
+
+def generateUniqueTempFile( ext = '.txt' ):
+    """ Generate a unique file with a given filename with some random contents and return the name of the file on disk
+    Args:
+        ext (str): This is the extension (including '.') to give to the file of interest
+    """
+
+    with tempfile.NamedTemporaryFile(mode='w',suffix=ext,delete=False) as myFile:
+
+        myFile.write( ''.join(random.choice(string.ascii_uppercase+string.digits) for _ in range(20)) )
+
+        TestMassStorageClient._managed_files.append(myFile.name)
+
+        return myFile.name
+
+class TestMassStorageClient(GangaUnitTest):
+    """test for sjid in filename names explain each test"""
+
+    _managed_files = []
+
+    # Num of sj in tests
+    sj_len = 3
+
+    # Where on local storage we want to have our 'MassStorage solution'
+    outputFilePath = '/tmp/MassStorageClient'
+
+    # This sets up a MassStorageConfiguration which works by placing a file on local storage somewhere we can test using standard tools
+    MassStorageTestConfig = {'defaultProtocol': 'file://',
+                             'fileExtensions': [''],
+                             'uploadOptions': {'path': outputFilePath, 'cp_cmd': 'cp', 'ls_cmd': 'ls', 'mkdir_cmd': 'mkdir -p'},
+                             'backendPostprocess': {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client',
+                                                    'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}}
+
+    def setUp(self):
+        """
+        Configure the MassStorageFile for the test
+        """
+        extra_opts=[('PollThread', 'autostart', 'False'),
+                    ('Local', 'remove_workdir', 'False'),
+                    ('TestingFramework', 'AutoCleanup', 'False'),
+                    ('Output', 'MassStorageFile', copy.deepcopy(TestMassStorageClient.MassStorageTestConfig)),
```

Comment:
Ah, I was debugging why the options weren't being applied here. I can remove the deepcopy.

And no, unfortunately this can't be a parameter in the `setUp` function as the monitoring loop then starts with the default config and proceeds to start processing the completed jobs until it gets past the super call below.
(aka the monitoring loop is too fast!)
Tracking that down cost a lot of time but raises some concerns that some tests are passing incorrectly.